src/Authentication.Abstractions/AzureEnvironment.cs

```diff
@@ -44,30 +44,34 @@ public class AzureEnvironment : IAzureEnvironment, IEquatable<AzureEnvironment>
                 if (string.IsNullOrEmpty(armMetadataRequestUri))
                 {
                     armMetadataRequestUri = Environment.GetEnvironmentVariable(ArmMetadataEnvVariable);
-                    if (string.IsNullOrEmpty(armMetadataRequestUri))
-                    {
-                        armMetadataRequestUri = DefaultArmMetaDataEndpoint;
-                    }
-                    else
+                    if (!string.IsNullOrEmpty(armMetadataRequestUri))
                     {
                         debugLogger?.Invoke($"Get {armMetadataRequestUri} from environment variable {ArmMetadataEnvVariable}");
                     }
                 }
                 if (DisableArmMetaDataEndpoint.Equals(armMetadataRequestUri?.Trim().ToUpper()))
                 {
-                    warningLogger?.Invoke($"Discover feature is disabled by environment variable {ArmMetadataEnvVariable}");
+                    warningLogger?.Invoke($"Discover feature is disabled.");
                     armMetadataRequestUri = null;
                 }
                 if (!string.IsNullOrEmpty(armMetadataRequestUri))
                 {
                     debugLogger?.Invoke($"Discover environments via {armMetadataRequestUri}");
-                    List<ArmMetadata> list = InitializeEnvironmentsFromArm(armMetadataRequestUri, httpOperations).ConfigureAwait(false).GetAwaiter().GetResult();
-                    foreach (var metadata in list)
+                    try
                     {
-                        var env = MapArmToAzureEnvironment(metadata);
-                        env.Type = TypeDiscovered;
-                        armAzureEnvironments[env.Name] = env;
-                        debugLogger?.Invoke($"Added discovered environment {env.Name}");
+                        var list = InitializeEnvironmentsFromArm(armMetadataRequestUri, httpOperations).ConfigureAwait(false).GetAwaiter().GetResult();
+                        foreach (var metadata in list)
+                        {
+                            var env = MapArmToAzureEnvironment(metadata);
+                            env.Type = TypeDiscovered;
+                            armAzureEnvironments[env.Name] = env;
+                            debugLogger?.Invoke($"Added discovered environment {env.Name}");
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        warningLogger?.Invoke($"Cannot discover environments from Arm with Uri {armMetadataRequestUri}");
+                        warningLogger?.Invoke(e.StackTrace);
```

Comment:
We should be formal with strings visible to customers.
Also it's generally not a good idea to write a lot of warnings, especially when the error is not blocking.
```suggestion
                        warningLogger?.Invoke($"Cannot discover environments from ARM with URI {armMetadataRequestUri}. Falling back to built-in environments.");
                        debugLogger?.Invoke(e.StackTrace);
```